### PR TITLE
AB#1059 Add roles / permissions

### PR DESCRIPTION
**Pull request by daniel-weisse**
Explains user permissions, or more specifically, roles in the docs.
PR for the implementation in Marblerun: https://github.com/edgelesssys/marblerun/pull/189

### CODE REVIEW

File: content/docs/workflows/define-manifest.md [hunk 1]
```diff
@@ -308,6 +319,47 @@ Use the following command to preserve newlines correctly:
 awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;}' public_key.pem
 ```
 
+## Manifest:Roles
+
+The entry Roles defines roles which can be given to entities of Marblerun.
+Each role defines a `ResourceType` (one of `Packages` or `Secrets`), a list of `ResourceNames` of that type, and a list of `Actions` that role permitts on the listed resources. \
+Valid `Actions` are:
```
Comment (m1ghtym0) [inline content/docs/workflows/define-manifest.md:326]:
```suggestion
Marblerun supports Role-based access control (RBAC).
An RBAC Role contains rules that represent a set of permissions for the Marblerun entities `Users` and `Marbles`. Permissions are purely additive (there are no "deny" rules).
Each role defines a `ResourceType` (one of `Packages` or `Secrets`), a list of `ResourceNames` of that type, and a list of `Actions` that role permits on the listed resources. \
Valid `Actions` are:
```

File: content/docs/workflows/define-manifest.md [hunk 2]
```diff
@@ -308,6 +319,47 @@ Use the following command to preserve newlines correctly:
...
+                "ReadSecret",
+                "WriteSecret"
+            ]
+        }
+    }
+    //...
+}
+```
+{{<note>}}
+Manifest updates will only be possible if you created a role allowing updates to the neccessary packages, and provided at least one user with said role.
```
Comment (m1ghtym0) [inline content/docs/workflows/define-manifest.md:360]:
```suggestion
Deployment updates will only be possible if you create a role with permissions to update the particular packages and bind a user to that role.
```

File: content/docs/workflows/update-manifest.md [hunk 3]
```diff
@@ -11,7 +11,8 @@ In order to ensure the confidentiality of a deployed application, Marblerun uses
 Yet, updates play an important role to ensure your software stays secure. To avoid having to redeploy your application from scratch, Marblerun allows uploading a separate "Update Manifest" which increases the minimum `SecurityVersion` of one or multiple already deployed packages. After such an update is performed, an old version of a defined software package cannot be loaded anymore under the current Manifest.
 
 ## Requirements
-In order to deploy an Update Manifest, the original Manifest, you need to be in possession of a certificate/private key pair which has been defined in the `Users` section of the original Manifest, as described in ["Defining a Manifest"]({{< ref "docs/workflows/define-Manifest.md#manifestmarbles" >}}).
+In order to deploy an Update Manifest, you need to be in possession of a certificate/private key pair belonging to a user  from the `Users` section of the original Manifest, as described in ["Defining a Manifest"]({{< ref "docs/workflows/define-Manifest.md#manifestmarbles" >}}).
+Furthermore the user needs to be [permitted to update]({{< ref "docs/workflows/define-Manifest.md#manifestroles" >}}) the chosen packages.
```
Comment (m1ghtym0) [inline content/docs/workflows/update-manifest.md:15]:
```suggestion
In order to deploy an Update Manifest, you need to be in possession of a certificate/private key pair belonging to a user from the `Users` section of the original Manifest, as described in ["Defining a Manifest"]({{< ref "docs/workflows/define-Manifest.md#manifestmarbles" >}}).
Furthermore the user needs to be [permitted to update]({{< ref "docs/workflows/define-Manifest.md#manifestroles" >}}) the chosen packages.
```

File: content/docs/workflows/define-manifest.md [hunk 4]
```diff
@@ -250,21 +251,30 @@ The following gives some examples.
 * Inject a symmetric key in hex format: `{{ hex .Secrets.secret_aes_key }}`
 
 ## Manifest:Users
-The optional entry `Users` defines user credentials and permissions for authentication and access control.
+The optional entry `Users` defines user credentials and role-bindings for authentication and access control.
```
Comment (m1ghtym0) [inline content/docs/workflows/define-manifest.md:254]:
```suggestion
The optional entry `Users` defines user credentials and role bindings for authentication and access control.
```

**Review by m1ghtym0 — CHANGES_REQUESTED** [review]

**Comment by m1ghtym0** [comment]
Do we have role support for Marbles as well or only Users so far?

**Comment by daniel-weisse** [comment]
Currently its only for Users

**Pull request merged**